collaboration/sig-core/proposals/upgrade-to-knative-mesh.md

```diff
@@ -0,0 +1,147 @@
+|`event-service` build| `event-service` providing backward compatibility|Docker image| 
+|`application` chart | `application` chart includes the new `event-service` docker image|[application](https://github.com/kyma-project/kyma/tree/master/components/application-operator/charts/application) new chart version| 
+|`application-operator` build| A new `application-operator` build that includes the new `application` chart version in previous step|Docker image| 
+|`application-connector` chart |`application-connector` new chart version with the updated `application-operator`|[application-connector](https://github.com/kyma-project/kyma/tree/master/resources/application-connector) new chart version|
+|`application-connector` `post-upgrade` hook| A `post-upgrade` k8s job that checks if all applications event-services were upgraded correctly and fails if otherwise |`application-connector` new version| 
+
+
+The new `Event Service` should serve legacy endpoint `/v1/events` and route them to the proper event mesh `HttpSource` adapter.
+
+#### Execution logic
+
+- Once the `application-connector` gets upgraded, the new `application-operator` will be start up.
+- Each `Application` has a helm release with the application name (e.g. `commerce`). 
+- The new `application` chart includes the new `event-service` deployment.
+- When starting, the `application-operator` will compare the `application` chart version in its folder with the version of each release (e.g. `commerce-prod`, `varkes`, etc ) and upgrade it if needed.
+- After `application` releases are upgraded the new `event-service` pods will restart and the new ones will start serving the legacy endpoint and routing the events to the HttpSource.
+- The `post-upgrade` Job of the `application-connector` chart should wait, check for failure conditions and fail if any of them are true.
+ 
+#### Failure conditions 
+The `post-upgrade` job should fail when
+
+- Any of the `event-service` instances are not updated
+- Any of the `event-service` instances are failing
+
+#### III. User namespace migration
+
+#### Changes summary
+
+| Name | Description | Artifact |
+|------|-------------|----------|
+|`ServiceInstance` recreation binary| A Go binary which creates `Trigger` objects for each Kyma `Subscription` objects | Docker image|
+|`Trigger` sync binary| A Go binary which recreates `ServiceInstance` objects  | Docker image|
+|`core` `pre-upgrade` hooks | Two `pre-upgrade` Jobs which will be executed when Kyma installer upgrades the `core` chart |`core` new version |
+   
+#### Execution logic
+
+- Recreate each `ServiceInstance` object in its same namespace (check CBS code [here](https://github.com/kyma-project/kyma/blob/master/components/console-backend-service/internal/domain/servicecatalog/serviceinstance_service.go#L239)).
+- Find all Kyma `Susbscription` objects and create a matching trigger object.
+- Delete each `Susbscription` object after a matching trigger is successfully created.  
```

Comment:
Does it makes sense to mention, that the deletion of a *Subscription* CR has to be done if and only if the subscription controller is either scaled down to 0 or completely removed from the kyma-system.

Reply:
My Bad, I was thinking about the knative subscriptions which are reconciled by the subscription controller.